composition-js/src/merging/merge.ts

```diff
@@ -826,20 +859,34 @@ class Merger {
       if (!source) {
         continue;
       }
-      if (source.hasAppliedDirective('key')) {
+      const keyDirectiveName = federationMetadata(source.schema())?.keyDirective().name ?? 'key';
+      if (source.hasAppliedDirective(keyDirectiveName)) {
         sourceAsEntity.push(source);
       } else {
         sourceAsNonEntity.push(source);
       }
     }
+    sources.forEach((source, idx) => {
```

Comment:
I believe this block of code does the same than the one before that. I assume you decided to rewrote it and forgot to remove the old version? I do like the new version better.

Reply:
yup, exactly right.